client/src/main/java/io/pravega/client/stream/impl/ClientFactoryImpl.java

```diff
@@ -178,15 +179,32 @@ public ClientFactoryImpl(String scope, Controller controller, ConnectionFactory
         Consumer<Segment> segmentSealedCallBack = s -> {
             throw new IllegalStateException("RevisionedClient: Segmentsealed exception observed for segment:" + s);
         };
-        String delegationToken = Futures.getAndHandleExceptions(controller.getOrRefreshDelegationTokenFor(segment.getScope(),
-                segment.getStreamName()), RuntimeException::new);
+
+        String delegationToken = checkStreamReturnDelegation(streamName, segment);
+
         SegmentOutputStream out = outFactory.createOutputStreamForSegment(segment, segmentSealedCallBack,
                 config.getEventWriterConfig(), delegationToken);
         ConditionalOutputStream cond = condFactory.createConditionalOutputStream(segment, delegationToken, config.getEventWriterConfig());
         SegmentMetadataClient meta = metaFactory.createSegmentMetadataClient(segment, delegationToken);
         return new RevisionedStreamClientImpl<>(segment, in, out, cond, meta, serializer);
     }
 
+    private String checkStreamReturnDelegation(String streamName, Segment segment) {
+        CompletableFuture<Stream.State>  streamState = controller.getStreamState(scope, streamName);
+        CompletableFuture<String> delegationToken = controller.getOrRefreshDelegationTokenFor(segment.getScope(), segment.getStreamName());
+        return streamState
+                .thenCombine(delegationToken,
+                        (state, delTok) -> {
+                            if (state != Stream.State.UNKNOWN && state != Stream.State.SEALED) {
+                                return delTok;
+                            } else {
+                                log.warn("Stream does not exist.");
```

Comment:
This log message and exception would confuse someone who's stream is simply sealed. 
An appropriate message should be logged or thrown. (Maybe include the stream name and the attempted operation)

Reply:
removed